src/util/heap/monotonepageresource.rs

```diff
@@ -28,9 +28,9 @@ pub struct MonotonePageResource<VM: VMBinding> {
 
 struct MonotonePageResourceSync {
     /** Pointer to the next block to be allocated. */
-    cursor: Address,
+    pub cursor: Address,
```

Comment:
These fields do not need to be public. From what I can see, you are not using them outside this module.